cmake/Platform/Common/Clang/Configurations_clang.cmake

```diff
@@ -51,9 +51,14 @@ ly_append_configurations_options(
         -g # debug symbols
     COMPILATION_RELEASE
         -O2
-    LINK_NON_STATIC
-        -Wl,-undefined,error
 )
 
+if(NOT CMAKE_LINKER MATCHES ".*lld-link\.exe$")
+    ly_append_configurations_options(
+        LINK_NON_STATIC
+            -Wl,-undefined,error
+    )
+endif()
+
```

Comment:
So this is a non-Windows specific linker warning declaration.

Reply:
After looking into this a bit more it seems that `-Wl,-undefined,error` doesn't work on Linux (by which I mean, it doesn't cause a link-time failure if there are undefined symbols). This appears to be an accidental regression introduced in c02345fd from #1436, I'll amend this accordingly.

Reply:
I've restored the proper flags for the Mac/Linux specific cases, instead of exempting the flag for `lld-link.exe`.